### ISSUE TITLE
S3 Watcher dies when bucket lacks ACL.

### ISSUE THREAD

**Issue by charlesgentry** [Netflix/security_monkey#579]
I set up several AWS accounts with successful scanning, but one account keeps bombing out at the same point and never shows in the "Account" box on the "Search" page. 
Fresh install as of today March 1 2017

last part of the securitymonkey,log
```
2017-03-02 00:07:08,883 DEBUG: Slurping S3 Bucket (XXXXXXXX) from 123412341234 [in /usr/local/src/security_monkey/security_monkey/watchers/s3.py:48]
2017-03-02 00:07:08,932 DEBUG: Logging exception from scheduler-change-reporter-uncaught with location: None to the database. [in /usr/local/src/security_monkey/security_monkey/datastore.py:636]
2017-03-02 00:07:08,938 DEBUG: Completed logging exception to database. [in /usr/local/src/security_monkey/security_monkey/datastore.py:668]
2017-03-02 00:07:08,938 ERROR: Job "run_change_reporter (trigger: interval[0:15:00], next run at: 2017-03-02 00:18:09.775527)" raised an exception [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:520]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/apscheduler/scheduler.py", line 512, in _run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/usr/local/src/security_monkey/security_monkey/scheduler.py", line 32, in run_change_reporter
    reporter.run(account, interval)
  File "/usr/local/src/security_monkey/security_monkey/reporter.py", line 53, in run
    unique_watcher_results = self.slurp_unique_watchers(mons, account, interval)
  File "/usr/local/src/security_monkey/security_monkey/reporter.py", line 97, in slurp_unique_watchers
    (items, exception_map) = monitor.watcher.slurp()
  File "/usr/local/src/security_monkey/security_monkey/watchers/s3.py", line 66, in slurp
    return slurp_items()
  File "/usr/local/src/security_monkey/security_monkey/decorators.py", line 151, in decorated_function
    itm, exc = f(*args, **kwargs)
  File "/usr/local/src/security_monkey/security_monkey/watchers/s3.py", line 62, in slurp_items
    item = S3Item.from_slurp(bucket_name, bucket, **kwargs)
  File "/usr/local/src/security_monkey/security_monkey/watchers/s3.py", line 84, in from_slurp
    region=bucket['Region'],
KeyError: 'Region'
```

**Comment by monkeysecurity** [comment]
Could you do a `pip freeze | grep cloudaux` and let me know which version you have?

Latest is 1.1.2.

Looks like you're on the master branch from the line numbers.

May be worthwhile to edit s3.py to have it print bucket before it bombs.

@mikegrima - Any ideas?

**Comment by mikegrima** [comment]
That's odd... Which region is this bucket in @charlesgentry ?

**Comment by charlesgentry** [comment]
cloudaux==1.1.2

The last bucket scanned before the error is in us-east-1

I found another clue...
when I try to look at the bucket in the console with MY User account, which has "AdministratorAccess" permissions (allow * on *), I get:

"Sorry! You do not have permissions to view this bucket."

I can't do anything with the last bucket that gets scanned before the error.

**Comment by monkeysecurity** [comment]
Yeah - that will happen if someone deletes the bucket ACL and the bucket policy doesn't have anything allowing you access.

You'll need to signin as ROOT to the AWS account to fix the bucket.

Still, we don't want security_monkey dying on these.  I'll keep this open to track progress.

**Comment by monkeysecurity** [comment]
Fixed in #633. Closing.
